core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2023,6 +2023,22 @@ package object config {
       .stringConf
       .createWithDefaultString("PWR")
 
+  private[spark] val EXECUTOR_DECOMMISSION_BATCH_INTERVAL =
+    ConfigBuilder("spark.executor.decommission.batchInterval")
+      .doc("Executors are decommissioned in batches to avoid overloading network bandwidth in" +
+        " migrating rdd and shuffle data. This config sets the interval between batches.")
+      .version("3.2.0")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefault(3000)
+
+  private[spark] val EXECUTOR_DECOMMISSION_BATCH_SIZE =
```

Comment:
Is there any way to reasonably pick good values for these configuration options?

The load issues caused by deallocation are only indirectly related to the number of executors, because the amount of RDD transfer work needed is different for each job and each executor within each job depending on how many tasks it ran or numerous other factors.

Reply:
I think its hard to do well here.

In a perfect case scenario we could introspect the block manager and shuffle manager since they know where data is and configure a maximum number of blocks, but I'm of the belief that perfect is the enemy of done and I _think_ this could be an OK approximation, especially as we add better logic to pick which executors to decommission.